apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -32,6 +34,28 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
     {:ok, nil}
   end
 
+  defp maybe_move_cursor_to_next(type, document, line, column)
+       when type in [:function, :module, :macro] do
+    position = Position.new(document, line, column)
+
+    case Ast.zipper_at(document, position) do
+      {:ok, zipper} ->
+        zipper = %{node: {entity_name, meta, _}} = Sourceror.Zipper.next(zipper)
+
+        if entity_name == :when do
+          %{node: {_entity_name, meta, _}} = Sourceror.Zipper.next(zipper)
+          {meta[:line], meta[:column]}
+        else
+          {meta[:line], meta[:column]}
+        end
```

Comment:
```suggestion
    meta = 
        if entity_name == :when do
          %{node: {_entity_name, meta, _}} = Sourceror.Zipper.next(zipper)
          meta
        else
          meta
        end
    {meta[:line], meta[:column]}
```
Though why don't you use the helper you defined to extract the position from the metadata?

Reply:
Applied.

> Though why don't you use the helper you defined to extract the position from the metadata?

Because this is the definition module, there is only one place where the position is extracted.

Reply:
The helper i'm talking about is the one that extracts range info from the metadata.

Reply:
So, do I still need to make some modifications?